src/wh_server_crypto.c

```diff
@@ -79,6 +103,244 @@ static int wh_Server_HandleSharedSecretCurve25519(whServerContext* ctx,
 
 
 /** Public server crypto functions */
+#ifndef NO_RSA
+int wh_Server_CacheImportRsaKey(whServerContext* ctx, RsaKey* key,
+        whKeyId keyId, whNvmFlags flags, uint32_t label_len, uint8_t* label)
+{
+    int ret = 0;
+    uint8_t* cacheBuf;
+    whNvmMetadata* cacheMeta;
+    uint16_t max_size;
+    uint16_t der_size;
+
+    if (    (ctx == NULL) ||
+            (key == NULL) ||
+            (WH_KEYID_ISERASED(keyId)) ||
+            ((label != NULL) && (label_len > sizeof(cacheMeta->label)))) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* wc_RsaKeyToDer doesn't have a length check option so we need to just pass
+     * the big key size if compiled */
+    /* TODO: Change this to use an estimate of the DER size based on key len */
+    if(WOLFHSM_CFG_SERVER_KEYCACHE_BIG_COUNT > 0) {
+        max_size = WOLFHSM_CFG_SERVER_KEYCACHE_BIG_BUFSIZE;
+    } else {
+        max_size = WOLFHSM_CFG_SERVER_KEYCACHE_BUFSIZE;
+    }
+
+    /* get a free slot */
+    ret = hsmCacheFindSlotAndZero(ctx, max_size, &cacheBuf, &cacheMeta);
+    if (ret == 0) {
+        ret = wh_Crypto_SerializeRsaKey(key, max_size, cacheBuf, &der_size);
+    }
+
+    if (ret == 0) {
+        /* set meta */
+        cacheMeta->id = keyId;
+        cacheMeta->len = der_size;
+        cacheMeta->flags = flags;
+        cacheMeta->access = WH_NVM_ACCESS_ANY;
```

Comment:
eventually we want `access` to be passed as an arg to all the import functions, right?

Reply:
yup.  and we need to return more of the keyid to the client.  right now, we ONLY return the ID portion, stripping off the higher bits.